lib/services/cloud-project-service.ts

```diff
@@ -87,26 +86,40 @@ export class CloudProjectService extends CloudService implements ICloudProjectSe
 
 		const tasksResults: IDictionary<IServerResult> = {};
 		this.$logger.info(`${EOL}### Build machines cleanup${EOL}`);
-		for (let i = 0; i < cleanupResponse.buildMachineResponse.length; i++) {
-			const taskId = `Cloud Cleanup Task #${i + 1}`;
-			const task = cleanupResponse.buildMachineResponse[i];
-			try {
-				await this.waitForServerOperationToFinish(taskId, task);
-				const taskResult = await this.getObjectFromS3File<IServerResult>(task.resultUrl);
-				const output = this.$nsCloudOutputFilter.filter(await this.getContentOfS3File(task.outputUrl));
-				if (this.hasContent(output)) {
-					this.$logger.info(output);
+		const childTasks: Promise<IChildTaskExecutionResult>[] = [];
+		for (let task of cleanupResponse.buildMachineResponse) {
+			const executeChildTask = async (): Promise<IChildTaskExecutionResult> => {
+				const childCloudOperationId = task.cloudOperationId;
+				let output = `Child cloud operation id: ${childCloudOperationId}`;
+				try {
+					const taskResult = await this.waitForServerOperationToFinish(childCloudOperationId, task, { silent: true });
+					tasksResults[childCloudOperationId] = taskResult;
+					output += await this.getCollectedLogs(task);
+					return { output };
+				} catch (err) {
+					if (this.getResult(childCloudOperationId)) {
+						tasksResults[childCloudOperationId] = this.getResult(childCloudOperationId);
+					}
+
+					// We don't want to stop the execution if one of the tasks fails.
+					return { output, err: `${childCloudOperationId} error: ${err}` };
 				}
+			};
+
+			childTasks.push(executeChildTask());
+		}
+
+		for (let task of childTasks) {
+			const taskResult = await task;
+			this.$logger.info(taskResult.output);
```

Comment:
Why not print the output and the errors in the previous FOR cycle?

Reply:
Because we need to start all cleanup tasks and then wait for their results and print their logs. If we wait the result of each cleanup task in the previous FOR cycle, the handshake between the build machine and the client application may fail for some tasks for cloud operation v2. Also if we print the logs there, we will receive mixed output from all tasks for cloud operation v2.

Reply:
Because the initial logic was not `{ silent: true }`. Now we can print the result in the first FOR cycle.